compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/ProjectLoader.java

```diff
@@ -95,31 +136,41 @@ public static Project loadProject(Path path, ProjectEnvironmentBuilder projectEn
      * @return documentId of the document
      */
 
-    public static Optional<DocumentId> getDocumentId(Path documentFilePath, BuildProject project) {
+    public static Document getDocument(Path documentFilePath, Project project) {
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            DocumentId documentId = project.currentPackage().getDefaultModule().documentIds().iterator().next();
+            return project.currentPackage().getDefaultModule().document(documentId);
+        }
+        if (!(project instanceof BuildProject)) {
```

Comment:
Since this method is only supported for SINGLE_FILE_PROJECT and BUILD_FILE_PROJECT, we can have a top level check to throw 'unsupported project type' to avoid `instanceof` check

Reply:
We don't need this `instanceof` check. One should be using the `ProjectKind`